src/services/moderation.service.ts

```diff
@@ -331,6 +318,110 @@ export class ModService {
+    const id = await Moderation.Helpers.resolveUser(guild, user_handle);
+    if (!id) {
+      throw new Error('User not found');
+    }
+
+    const modreports = collections?.modreports;
+    const modwarnings = collections?.modwarnings;
+
+    const reports = await modreports?.find({ guild: guild.id, user: id }).toArray();
+    const warnings = await modwarnings?.find({ guild: guild.id, user: id }).toArray();
+    const banStatus = await this._getBanStatus(collections, guild, id);
+
+    if (!reports) {
+      throw new Error('Couldnt get reports');
+    }
+
+    //Number of Reports > warns
+    //Each row has 2 cells, left cell is report, right cell is warn
+    const rows: string[][] = new Array(reports.length);
+    reports.forEach((report, i) => {
+      rows[i] = new Array(2);
+      rows[i][0] = this._serializeReportForTable(report);
+
+      const reportID = report._id?.toHexString();
+      if (!reportID || !warnings) {
+        return;
+      }
+
+      const relatedWarn = warnings.filter((w) => w.reportId?.toHexString() === reportID);
+      if (!relatedWarn?.length) {
+        return;
+      }
+
+      rows[i][1] = this._serializeWarningForTable(relatedWarn[0]);
+    });
+
+    //Create HTML table
+    const table = this._createTableFromReports(rows);
+
+    //Retrieve template
+    const defaultHTML = fs.readFileSync(
+      './src/app/plugins/__generated__/reportTemplate.html',
+      'utf8'
+    );
+
+    //Replace the placeholders with data we've collected
+    const data = defaultHTML
+      .replace('BAN_STATUS', banStatus)
+      .replace('DYNAMIC_TABLE', table)
+      .replace('NUM_REPORTS', reports.length + '')
+      .replace('NUM_WARNS', warnings?.length + '' || '0')
+      .replace('USER_NAME', user_handle);
+    return await this._writeDataToFile(data);
+  }
+
+  private async _getBanStatus(collections: any, guild: Guild, id: string): Promise<string> {
+    const modbans = collections?.modBans;
+    const bans = await modbans?.find({ guild: guild.id, user: id });
+
+    const mostRecentBan =
+      (await bans
+        ?.sort({ date: -1 })
+        .limit(1)
+        .toArray()) || [];
+
+    if (mostRecentBan.length && mostRecentBan[0].active) {
+      return `Banned since ${mostRecentBan[0].date.toLocaleString()}`;
+    }
+    return 'Not banned';
+  }
+
+  private _createTableFromReports(rows: string[][]) {
+    //Wrap each cell in <td> tags
+    //Wrap each row in <tr> tags
+    return rows
+      .map((row: string[]) => `<tr>${row.map((cell) => `<td>${cell}</td>`).join('\n')}</tr>`)
+      .join('\n');
+  }
+
+  private _serializeReportForTable(report: Moderation.IModerationReport): string {
+    const ret = `Reported on: ${report.timeStr}<br />Description: ${report.description ||
+      'No Description'}`;
+    if (!report.attachments?.length) {
+      return ret;
+    }
+
+    return `${ret}<br />Attachment: <img src="${report.attachments[0]}">`;
```

Comment:
there can actually be multiple attachments and they might not be images, fyi

yikes that makes it kinda annoying... maybe just see if it ends in .png or .jpg, or just dump all the links